db/migrate/20230221070501_create_customer_metadata.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class CreateCustomerMetadata < ActiveRecord::Migration[7.0]
+  def change
+    create_table :customer_metadata, id: :uuid do |t|
+      t.references :customer, type: :uuid, null: false, foreign_key: true, index: true
+
+      t.string :key, null: false
```

Comment:
What about a unique index on both `customer_id` and `key` fields to match the uniqueness validation?

Reply:
Make sense to add it on DB level as well! Added!